avaframe/ana4Stats/probAna.py

```diff
@@ -93,30 +92,48 @@ def updateCfgRange(cfg1, cfgProb, varName):
+            message = ('Only one reference value is allowed for %s: but %s is given' %
+                (varPar, cfg1['GENERAL'][varPar]))
             log.error(message)
             raise AssertionError(message)
-        if varPar == 'relTh':
-            cfg1['GENERAL']['useRelThFromIni'] = 'True'
-        elif varPar == 'entTh':
-            cfg1['GENERAL']['useEntThFromIni'] = 'True'
+        elif varPar in ['entTh', 'relTh', 'secondaryRelTh']:
+            thPercentVariation = varPar + 'PercentVariation'
+            if cfg1['GENERAL'][thPercentVariation] != '':
+                message = ('Only one reference value is allowed for %s: but %s %s is given' %
+                    (varPar, thPercentVariation, cfg1['GENERAL'][thPercentVariation]))
+                log.error(message)
+                raise AssertionError(message)
 
     # get range, steps and reference value of parameter to perform variations
-    valRange = cfgProb['PROBRUN'].getfloat('%sRange' % varName)
-    valSteps = cfgProb['PROBRUN'].getint('%sSteps' % varName)
-    valVal = cfg1['GENERAL'].getfloat(varName)
+    valVariation = cfgProb['PROBRUN']['%sVariation' % varName]
+    valSteps = cfgProb['PROBRUN']['%sSteps' % varName]
+    valVal = cfg1['GENERAL'][varName]
 
     # set variation in configuration
-    cfg1['GENERAL'][varName] = '%s:%s:%s' % (str(valVal-valRange), str(valVal+valRange), str(valSteps))
+    if varName in ['relTh', 'entTh', 'secondaryRelTh']:
+        if cfgProb['PROBRUN'].getboolean('percentVariation'):
+            parName = varName + 'PercentVariation'
+            cfg1['GENERAL'][parName] = valVariation + '$' + valSteps
+        else:
+            parName = varPar + 'rangeVariation'
+            cfg1['GENERAL'][parName] = valVariation + '$' + valSteps
+        cfg1['GENERAL']['addStandardConfig'] = 'True'
+    else:
+        # set variation
+        if cfgProb['PROBRUN'].getboolean('percentVariation'):
+            cfg1['GENERAL'][varName] = '%s$%s$%s' % (valVal, valVariation, valSteps)
+            valValues = fU.splitIniValueToArraySteps(cfg1['GENERAL'][varName])
+        else:
+            valStart = str(float(valVal) - float(valVariation))
+            valStop = str(float(valVal) + float(valVariation))
+            cfg1['GENERAL'][varName] = '%s:%s:%s' % (valStart, valStop, valSteps)
+            valValues = np.linspace(float(valStart), float(valStop), int(valSteps))
 
-    # if reference value is not in this list - add reference values
-    valValues = np.linspace(valVal-valRange, valVal+valRange, valSteps)
-    refIn = False
-    if valVal not in valValues:
-        cfg1['GENERAL'][varName] = '%s:%s:%s&%s' % (str(valVal-valRange), str(valVal+valRange), str(valSteps), str(valVal))
-        refIn = True
+        # if reference value is not in variation - add reference values
+        if valVal not in valValues:
+            cfg1['GENERAL'][varName] = cfg1['GENERAL'][varName] + '&' + str(valVal)
```

Comment:
Break this out to an extra function?

Reply:
I would suggest to keep this in one because this second part here just sets the actual values after first checking if no invalid settings are used and building the parameters